### PR TITLE
Vsphere dis migrate

### PR DESCRIPTION
**Pull request by sgaoshang**
Run job on IBMC devqe disconnected network

### CODE REVIEW

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-vsphere-discon-proxy

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-vsphere-discon-proxy

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-vsphere-discon-proxy

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-vsphere-discon-proxy

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-vsphere-discon-proxy

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-vsphere-discon-proxy

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-vsphere-discon-proxy

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-vsphere-discon-proxy

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-vsphere-discon-proxy

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-vsphere-discon-proxy

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-vsphere-discon-proxy

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-vsphere-discon-proxy

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-vsphere-discon-proxy

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-vsphere-discon-proxy

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-vsphere-discon-proxy

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-vsphere-discon-proxy

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-vsphere-discon-proxy

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-vsphere-discon-proxy

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-vsphere-discon-proxy

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-vsphere-discon-proxy

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-vsphere-discon-proxy

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-vsphere-discon-proxy

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-vsphere-discon-proxy

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-vsphere-discon-proxy

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-vsphere-discon-proxy

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-vsphere-discon-proxy

**Comment by sgaoshang** [comment]
@jianlinliu @jinyunma pls help review it, thanks.

File: ci-operator/step-registry/ipi/conf/vsphere/check/ipi-conf-vsphere-check-commands.sh [hunk 1]
```diff
@@ -40,7 +40,7 @@ if [ ${LEASE_NUMBER} -ge 151 ] && [ ${LEASE_NUMBER} -lt 160 ]; then
 fi
 
 # For leases >= 200, run on the IBM Cloud vSphere 8 env
-if [ ${LEASE_NUMBER} -ge 200 ]; then
+if [ ${LEASE_NUMBER} -ge 200 ] && [ ${LEASE_NUMBER} -lt 220 ]; then
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/vsphere/check/ipi-conf-vsphere-check-commands.sh:43]: The condition did not include the lease == 220, shall we include the 220 here?

File: ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installer-rehearse-4.14.yaml [hunk 2]
```diff
@@ -96,6 +96,16 @@ tests:
   steps:
     cluster_profile: ibmcloud-qe
     workflow: cucushift-installer-rehearse-ibmcloud-ipi-proxy-private
+- as: installer-rehearse-vsphere-discon-proxy
+  cron: '@yearly'
+  steps:
+    cluster_profile: vsphere-dis
+    workflow: cucushift-installer-rehearse-vsphere-ipi-proxy
+- as: installer-rehearse-vsphere-discon-upi
```
Comment (jianlinliu) [inline ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installer-rehearse-4.14.yaml:104]: I think we only need to add one `installer-rehearse-vsphere-discon` job here for a simple testing, wdyt?

File: ci-operator/step-registry/ipi/conf/vsphere/check/ipi-conf-vsphere-check-commands.sh (hunk 1 above)
Comment (sgaoshang) [reply ci-operator/step-registry/ipi/conf/vsphere/check/ipi-conf-vsphere-check-commands.sh:43]: Currently
poc vSphere 8 have segments from 200 to 213
devqe vSphere have segments 221 to 234
I'm OK with setting ${LEASE_NUMBER} -lt 221

File: ci-operator/step-registry/cucushift/agent/vsphere/disconnected/provision/cucushift-agent-vsphere-disconnected-provision-chain.yaml [hunk 3]
```diff
@@ -22,7 +22,7 @@ chain:
       documentation: |-
         Register DNS for mirror registry.
     - name: BASTION_HOST_SUBNET
-      default: "qe-segment"
+      default: "devqe-segment-221"
```
Comment (jinyunma) [inline ci-operator/step-registry/cucushift/agent/vsphere/disconnected/provision/cucushift-agent-vsphere-disconnected-provision-chain.yaml:25]: as you already change `BASTION_HOST_SUBNET` to segment on new vsphere, I guess that cluster profile also needs to be adjusted to `vsphere-dis` in relevant job and job config?

Comment (sgaoshang) [reply ci-operator/step-registry/cucushift/agent/vsphere/disconnected/provision/cucushift-agent-vsphere-disconnected-provision-chain.yaml:25]: Yes, I'll move all job with profile vsphere-discon to vsphere-dis

File: ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installer-rehearse-4.14.yaml (hunk 2 above)
Comment (sgaoshang) [reply ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installer-rehearse-4.14.yaml:104]: yes, I can remove one

File: ci-operator/step-registry/cucushift/agent/vsphere/disconnected/provision/cucushift-agent-vsphere-disconnected-provision-chain.yaml (hunk 3 above)
Comment (jinyunma) [reply ci-operator/step-registry/cucushift/agent/vsphere/disconnected/provision/cucushift-agent-vsphere-disconnected-provision-chain.yaml:25]: okay, after that, I think we can also verify the workflow `cucushift-installer-rehearse-vsphere-upi-disconnected-workers-rhcos-rhel8` in this PR, but need to update `RHEL_IMAGE` to `qe-rhel-88-release` in that install chain.

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-vsphere-discon-upi

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-devpreview-vsphere-upi-disconnected-sdn-workers-rhcos-rhel8-p2-f14

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-devpreview-vsphere-upi-disconnected-sdn-workers-rhcos-rhel8-p2-f14

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-devpreview-vsphere-upi-disconnected-sdn-workers-rhcos-rhel8-p2-f14

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-devpreview-vsphere-upi-disconnected-sdn-workers-rhcos-rhel8-p2-f14

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-vsphere-upi-disconnected-sdn-workers-rhcos-rhel8-p2-f14

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-vsphere-upi-disconnected-sdn-workers-rhcos-rhel8-p2-f14

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-vsphere-upi-disconnected-sdn-workers-rhcos-rhel8-p2-f14

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-devpreview-vsphere-upi-disconnected-sdn-workers-rhcos-rhel8-p2-f14

File: ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installer-rehearse-4.14.yaml (hunk 2 above)
Comment (jianlinliu) [reply ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installer-rehearse-4.14.yaml:99]: @sgaoshang suggest to update the job name to `installer-rehearse-vsphere-dis`

Comment (sgaoshang) [reply ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installer-rehearse-4.14.yaml]: Updated, thanks.

**Comment by jianlinliu** [comment]
cc @mhanss to review agent part. cc @liangxia for other parts.

**Comment by mhanss** [comment]
Agent part is good to go.

**Pull request merged**